arangod/Agency/Node.cpp

```diff
@@ -1196,6 +1196,16 @@ void Node::clear() {
   }
 }
 
+std::vector<std::string> Node::keys() const {
+  std::vector<std::string> result;
+  if (!_isArray) {
+    for (auto const& i : _children) {
```

Comment:
potential performance optimization:
```suggestion
    result.reserve(_children.size());
    for (auto const& i : _children) {
```